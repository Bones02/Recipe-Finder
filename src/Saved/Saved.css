@media only screen 
and (min-device-width: 300px)   
and (max-device-width: 600px) { 
    .saved_recipes {
        width: 100%;
        padding: 12px 20px;
        margin: 8px 0;
        display: inline-block;
        border: 1px solid grey;
        border-radius: 12px;
        text-align: center;
        background: #D3D3D3;
    }
    .saved_recipe_card {
        align-items: center;
        margin: auto;
        padding: 0px;
    }
    .saved_recipe_card li {
        width: 75%;
        padding:12px;
        margin:6px auto;
        background:#D3D3D3;
        text-align: center;
        border: 3px black solid;
        list-style-type: none;
    }
    .saved_recipe_card img {
        height: 100px;
        width: 100px;
    }
    .title {
        font: 16px;
        font-weight: bold;
    }
    .recipe_delete {
        margin-top: 20px;
    }
}
@media only screen 
and (min-device-width: 601px) { 
    .saved {
        margin: auto;
        justify-content: center;
    }
    .saved_recipes {
        width: 100%;
        padding: 12px 20px;
        margin: 8px 0;
        display: inline-block;
        border: 1px solid grey;
        border-radius: 12px;
        text-align: center;
        background: #D3D3D3;
    }
    .saved_recipe_card {
        justify-content: center;
        margin: auto;
        padding: 0px;
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        justify-content: center;
    }
    .saved_recipe_card li {
        width: 30%;
        height: 400px;
        padding:14px;
        margin:12px;
        background:#D3D3D3;
        text-align: center;
        border: 3px black solid;
        list-style-type: none;
        display: inline-block;
        justify-content: space-between;
    }
    .saved_recipe_card img {
        height: 100px;
        width: 100px;
    }
    .title {
        font: 16px;
        font-weight: bold;
        height: 80px;
    }
    .recipe_delete {
        margin-top: 20px;
    }
}