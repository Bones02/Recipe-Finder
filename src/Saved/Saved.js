import React, { Component } from "react";
import RecipeContext from '../RecipeContext';
//import './Saved.css'

class Saved extends Component {
    static defaultProps = {
        match: {
          params: {}
        }
    }

    static contextType = RecipeContext;

    render(){
        const { savedRecipes=[] } = this.context
        console.log(savedRecipes)
        return (
            <div>
                <p>
                    Saved Recipes
                </p>
                <p>
                    {savedRecipes}
                </p>
            </div>
            
            // <div>
            //     <ul className="recipe_card">
            //         {this.state.savedRecipes.map(recipe => (
            //         <li key={name}>
            //             <p>Id: {id}</p> 
            //             <p>Title: {title}</p> 
            //             <p>Serves: {servings}</p>
            //             <p>Ready in: {readyInMinutes} Minutes</p>
            //             <img src={`https://spoonacular.com/recipeImages/${image}`}/>   
            //             <button type="submit" className="button-delete" 
            //                 onClick={e => this.context.removeRecipe(recipe)}>Remove</button>  
            //         </li> 
            //         ))}
            //     </ul>
            // </div>
        )
    }
}


export default Saved;