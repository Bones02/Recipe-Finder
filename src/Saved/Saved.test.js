import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Saved from './Saved'

describe(`Saved component`, () => {
  it('renders a .saved by default', () => {
    const wrapper = shallow(<Saved />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})