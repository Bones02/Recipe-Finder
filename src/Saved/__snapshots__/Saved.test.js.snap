// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saved component renders a .saved by default 1`] = `
<section
  className="saved"
>
  <h2
    className="saved_recipes"
  >
    Saved Recipes
  </h2>
  <ul
    className="saved_recipe_card"
  />
</section>
`;
